docs/diagrams/ActivityManagerClassDiagram.puml

```diff
@@ -0,0 +1,89 @@
+@startuml
+skinparam classAttributeIconSize 0
+class ExerciseManager{
++ execute(parser : Parser) : String
+- executeEditAction(parser : Parser) : String
++ executeListAction(parser : Parser) : String
+- executeDeleteAction(parser : Parser) : String
+- executeAddAction(parser : Parser) : String
++ processAddExercise(parser : Parser) : Exercise
++ retrieveExercise(parser : Parser) : Exercise
++ processEditExercise(parser : Parser) : String
++ getActivityType(plural : boolean) : String
+- executeSearchAction(parser : Parser) : String
+}
+ExerciseManager --|> ActivityManager
+class WorkoutLogsManager{
++ execute(parser : Parser) : String
++ addWorkoutLog(workoutLogDate : String, workoutName : String) : void
++ addExerciseLog(workoutLogDate : String, exerciseName : String, weight : String, sets : String, repetitions : String) : void
++ getWorkoutLogString(date : String, workoutHashSet : HashSet<Exercise>) : String
++ getActivityType(plural : boolean) : String
++ exportToJSON() : JSONArray
+- {static} getWorkoutJson(exercises : HashSet<ExerciseLog>, workoutName : String, workoutDate : String) : JSONObject
+}
+WorkoutLogsManager --|> ActivityManager
+class WorkoutManager{
+- exerciseManager : ExerciseManager
++ WorkoutManager(exerciseManager : ExerciseManager)
++ execute(parser : Parser) : String
+- executeInfoAction(parser : Parser) : String
+- executeListAction(parser : Parser) : String
+- executeUnassignAction(parser : Parser) : String
+- executeAssignAction(parser : Parser) : String
+- executeDeleteAction(parser : Parser) : String
+- executeCreateAction(parser : Parser) : String
+- processWorkout(parser : Parser) : Workout
+- assignExerciseToWorkout(parser : Parser) : String
+- getFullWorkoutString(workoutPlanName : String) : String
+- unassignExerciseFromWorkout(parser : Parser) : String
++ getActivityType(plural : boolean) : String
+- executeSearchAction(parser : Parser) : String
+}
```

Comment:
Note that not all attributes and methods need be stated. Consider including only the relevant ones.